
# ğŸ§¬ Transcriptomics Pipeline (HISAT2 + StringTie + DESeq2)

This repository contains a reproducible pipeline for RNA-Seq transcriptomic analysis using **HISAT2** for alignment and **StringTie** for transcript assembly and quantification, and **DESeq2**
for differential gene expression analysis.

---

| Step | Description | Tool |
|:----:|:-------------|:------|
| 1ï¸âƒ£ | **Quality Control** | FastQC |
| 2ï¸âƒ£ | **Trimming** | Trim Galore|
| 3ï¸âƒ£ | **Alignment** | HISAT2 |
| 4ï¸âƒ£ | **Transcript Assembly** | StringTie |
| 5ï¸âƒ£ | **Quantification & Count Matrix Generation** | StringTie, prepDE.py |
| 6ï¸âƒ£ | **Differential Expression Analysis** | DESeq2 |
| 7ï¸âƒ£ | **Visualization & Interpretation** | R (ggplot2, pheatmap) |

---

## ğŸ§© Directory Structure

Transcriptomics-Pipeline/
â”‚
â”œâ”€â”€ liver/ # RNA-Seq analysis for liver tissue
â”‚ â”œâ”€â”€ data/ # Raw FASTQ files for liver
â”‚ â”‚ â”œâ”€â”€ Liver_sample.sh
â”‚ 
â”‚ â”œâ”€â”€ reference/ # Reference genome and annotation (liver)
â”‚ â”‚ â”œâ”€â”€ genome.fa  
â”‚ â”‚ â””â”€â”€ annotation.gtf
â”‚ â”‚
â”‚ â”œâ”€â”€ results/ # Output files from each step
â”‚ â”‚ â”œâ”€â”€ fastqc/ # Quality check reports
â”‚ â”‚ â”œâ”€â”€ merged/ # Merged assemblies
â”‚ â”‚ â””â”€â”€ deseq2/ # DESeq2 output
â”‚ â”‚
â”‚ â””â”€â”€ reads.sh # Pipeline script for liver dataset
â”‚
â”œâ”€â”€ kidney/ # RNA-Seq analysis for kidney tissue
â”‚ â”œâ”€â”€ data/
â”‚ â”‚ â”œâ”€â”€ kidney_samples.sh
â”‚ â”‚
â”‚ â”œâ”€â”€ reference/
â”‚ â”‚ â”œâ”€â”€ genome.fa
â”‚ â”‚ â””â”€â”€ annotation.gtf
â”‚ â”‚
â”‚ â”œâ”€â”€ results/ # Output files from each step
â”‚ â”‚ â”œâ”€â”€ fastqc/ # Quality check reports
â”‚ â”‚ â”œâ”€â”€ merged/ # Merged assemblies
â”‚ â”‚ â””â”€â”€ deseq2/ # DESeq2 output
â”‚ â”‚
â”‚ â””â”€â”€ reads.sh # Pipeline script for liver dataset
â”‚
â”œâ”€â”€ scripts/ # General or shared utility scripts
â”‚ â””â”€â”€ deseq2_analysis.R
â”‚
â”œâ”€â”€ envs/ # Conda environment files
â”‚ â””â”€â”€ transcriptomics.yaml
â”‚
â””â”€â”€ README.md


## ğŸ“Œ Dataset Details

- **Organism:** *Homo sapiens* (GRCh38)  
- **BioProjects:**  
  - Kidney â†’ [PRJNA93899] (https://www.ncbi.nlm.nih.gov/bioproject/PRJNA93899) # 36 single end samples
  - 
  - Liver â†’ [PRJNA750472](https://www.ncbi.nlm.nih.gov/bioproject/PRJNA750472)  # 12 single end samples
          â†’ [PRJNA429171](https://www.ncbi.nlm.nih.gov/bioproject/PRJNA429171)  # 6 paired end sample


## âœ¨ Summary of Findings

- RNA-Seq analysis revealed distinct **tissue-specific expression** in Marburg virusâ€“infected hosts.  
- **Liver:** 1074 upregulated, 187 downregulated genes.  
- **Kidney:** 972 upregulated, 363 downregulated genes.  
- **100 common DEGs** highlighted shared antiviral and immune responses.  
- **Hub genes** (*ANXA1*, *ATF3*, *KLF4*, *TLR2*) showed key regulatory roles.  
- *ATF3* and *ANXA1* interacted with viral proteins *VP30* and *VP35*, indicating immune modulation.  
- **Lariciresinol** emerged as a potential inhibitor with good ADMET properties.  
- Overall, the study identifies **key host targets and potential drug candidates** for Marburg virus infection.
