# Transcriptomics Pipeline (HISAT2 + StringTie + DESeq2)

This repository contains a reproducible pipeline for RNA-Seq Transcriptomic analysis using **HISAT2** for alignment and **StringTie** for transcript assembly and quantification, and **DESeq2** for differential gene expression analysis.

---

| Step | Description                                  | Tool                  |
| :--: | :------------------------------------------- | :-------------------- |
|  1Ô∏è‚É£ | **Quality Control**                          | FastQC                |
|  2Ô∏è‚É£ | **Trimming**                                 | Trim Galore           |
|  3Ô∏è‚É£ | **Alignment**                                | HISAT2                |
|  4Ô∏è‚É£ | **Transcript Assembly**                      | StringTie             |
|  5Ô∏è‚É£ | **Quantification & Count Matrix Generation** | StringTie, prepDE.py  |
|  6Ô∏è‚É£ | **Differential Expression Analysis**         | DESeq2                |
|  7Ô∏è‚É£ | **Visualization & Interpretation**           | R (ggplot2, pheatmap) |

---

## ‚öôÔ∏è Environment Setup & Pipeline Execution

Before running the pipeline, create the required Conda environment using the provided YAML file:

```bash
conda env create -f env.yaml
```

Once the environment is successfully created, run the main script:
 using arguements: $1 is samples folder name,
                   $2 is Output folder,
                   $3 is Reference genome folder.

```bash
./Transcriptomics_pipeline.sh <reads_folder> <output_folder> <reference_genome>
```

---

## üß´ Dataset Details

* **Organism:** *Homo sapiens* (GRCh38)
* **BioProjects:**

  * **Kidney:**

    * [PRJNA93899](https://www.ncbi.nlm.nih.gov/bioproject/PRJNA93899) ‚Äî 36 single-end samples
  * **Liver:**

    * [PRJNA750472](https://www.ncbi.nlm.nih.gov/bioproject/PRJNA750472) ‚Äî 12 single-end samples
    * [PRJNA429171](https://www.ncbi.nlm.nih.gov/bioproject/PRJNA429171) ‚Äî 6 paired-end samples

---

## üî¨ Biological Insights

* RNA-Seq analysis revealed distinct **tissue-specific expression** in Marburg virus‚Äìinfected hosts.
* **Liver:** 1074 upregulated, 187 downregulated genes.
* **Kidney:** 972 upregulated, 363 downregulated genes.
* **Common** DEGs highlighted shared antiviral and immune responses.
* **Hub genes** (*ANXA1*, *ATF3*, *KLF4*, *TLR2*) showed key regulatory roles.
* *ATF3* and *ANXA1* interacted with viral proteins *VP30* and *VP35*, indicating immune modulation.
* **Lariciresinol** emerged as a potential inhibitor with good ADMET properties.
* Overall, the study identifies **key host targets and potential drug candidates** for Marburg virus infection.
